### DISCUSSION TITLE
UTF-8 character in file name as UploadFile will raise "422 Unprocessable Entity"

### DISCUSSION THREAD

**Discussion by artiom-gesp** [fastapi/fastapi#8343] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

# App
```python
import uvicorn
from fastapi import FastAPI
from fastapi import File, UploadFile

app = FastAPI()

@app.post("/")
def read_root(file: UploadFile = File(...)):
    print(file.filename)
    return {"Hello": "World"}

uvicorn.run(app, host='0.0.0.0', port=8080)
```

# Request
```python
import os
import requests
import re

def rewrite_request(prepared_request):
    prepared_request.body = re.sub(b'filename\*.*\r', f'filename="{file_name}"\r'.encode() , prepared_request.body)
    return prepared_request

url="http://localhost:8080/"
file_name = '汉.txt'

with open(file_name, 'w+') as fd:
    fd.write('12345')

fp = open(file_name, 'rb')
files = {"file": (fp.name, fp)}

session = requests.Session()
response = session.post(url=url, files=files, auth=rewrite_request) # WORKS
response = session.post(url=url, files=files) # DOES NOT WORK
```

### Description

Hello,

I encounter the exact same issue as described here : https://github.com/tiangolo/fastapi/issues/1014.

When trying to upload a file containing special characters (such as chinese characters) in its name, FastAPI returns :
422 - Expected UploadFile, received: <class 'str'>.
I managed to find a workaround, but it's not practical (see code example).

### Operating System

Linux

### Operating System Details

Ubuntu 20.04

### FastAPI Version

0.78.0

### Python Version

3.8.13

### Additional Context

starlette version : 0.19.1

**Comment by acidjunk** [comment]
It could also point to an locale problem in the linux FS

What's the output of

```
locale
```

**Comment by artiom-gesp** [comment]
My apologies for the late reply. Here is the output of `locale` :
```
LANG=C.UTF-8
LANGUAGE=
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=
```

**Comment by mr-mateusz** [comment]
Hello,
Could you check your `urllib3` version and the raw request that is sent by requests (using `response.request.body`)?

I also have stumbled upon this issue, I have read about filename encoding in multipart and it seems to me that this issue is not related to FastApi at all.

`Urllib3` (which is used by requests under the hood) used to encode filenames using 'filename*=' (according to [RFC 2231](https://www.rfc-editor.org/rfc/rfc2231) and its extension [RFC 5987](https://www.rfc-editor.org/rfc/rfc5987)) but [RFC 7585](https://www.rfc-editor.org/rfc/rfc7578) made it obsolete. 

> NOTE: The encoding method described in [RFC5987], which would add a
>    "filename*" parameter to the Content-Disposition header field, MUST
>    NOT be used.


From version `1.25` `urllib3` encodes filename with 'filename='.

> Switched the default multipart header encoder from RFC 2231 to HTML 5 working draft. (Issue # 303, Pull # 1492)

https://github.com/urllib3/urllib3/blob/main/CHANGES.rst#125-2019-04-22

**Example**

```python
import requests

url = "http://localhost:8080/"
file_name = '汉.txt'

with open(file_name, 'w+') as fd:
    fd.write('12345')

fp = open(file_name, 'rb')
files = {"file": (fp.name, fp)}

response = requests.post(url, files=files)
print(response.text)
print(response.status_code)
print(response.request.body)
```
Output (`urllib3` `1.24.3`):
```
{"detail":[{"loc":["body","file"],"msg":"Expected UploadFile, received: <class 'str'>","type":"value_error"}]}
422
b'--89f219e8da6f67f3eccec7c2518e69bc\r\nContent-Disposition: form-data; name="file"; filename*=utf-8\'\'%E6%B1%89.txt\r\n\r\n12345\r\n--89f219e8da6f67f3eccec7c2518e69bc--\r\n'

```
Output (`urllib3` `1.26.12`):
```
{"Hello":"World"}
200
b'--1163be64eda3a1e6a6a5222aabd9ab53\r\nContent-Disposition: form-data; name="file"; filename="\xe6\xb1\x89.txt"\r\n\r\n12345\r\n--1163be64eda3a1e6a6a5222aabd9ab53--\r\n'
```

If you have `urllib3` < `1.25`, upgrade it. Then headers should be read correctly by FastApi.

**Reply by KajiMaCN** [reply]
Thanks for your help, it can work for me! @mr-mateusz

**Comment by tiangolo** [comment]
Thanks for the help here @mr-mateusz ! 👏  🙇 

If that solves the original problem, then you can close this issue @artiom-gesp  ✔️

**Comment by YuriiMotov** [comment, marked as answer]
Works fine in current version
